test/com/facebook/buck/android/aapt/MiniAaptTest.java

```diff
@@ -280,6 +280,28 @@ public void testVerifyReferences()
   }
 
   @Test
+  public void testInvalidNodeId() throws XPathExpressionException, IOException {
+    ImmutableList<String> lines = ImmutableList.<String>builder().add(
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+        "<LinearLayout>",
+        "<Button android:id=\"@+id/button1\" ",
+        "android:layout_toLeftOf=\"@button2\" />",
+        "</LinearLayout>")
+                                  .build();
+
+    Path resource = Paths.get("resource.xml");
+    filesystem.writeLinesToPath(lines, resource);
+
+    MiniAapt aapt = new MiniAapt(Paths.get("res"), Paths.get("R.txt"), ImmutableSet.<Path>of());
+    try {
+      aapt.processXmlFile(filesystem, resource, ImmutableSet.<RDotTxtEntry>builder());
+      fail("MiniAapt should throw parsing '@button2'");
+    } catch (ResourceParseException e) {
+      assertThat(e.getMessage(), containsString("Invalid definition of a resource"));
+    }
```

Comment:
Please use an `ExpectedException` instead (so that you don't need to the try/catch block).